### PR TITLE
fix: Mark deprecation on Channel & ChannelList modules

### PR DESCRIPTION
**Pull request by HoonBaek**
- Add deprecation mark on `Channel`, `ChannelProvider`, `ChannelList`, and `ChannelListProvider`

### CODE REVIEW

**Comment by HoonBaek** [comment]
Are there more places to add deprecation marks?

**Comment by bang9** [comment]
How about adding a migration guide link instead of providing an example code?

File: src/modules/Channel/context/ChannelProvider.tsx [hunk 1]
```diff
@@ -162,6 +162,12 @@ export interface ChannelProviderInterface extends ChannelContextProps, MessageSt
 
 const ChannelContext = React.createContext<ChannelProviderInterface | null>(null);
 
+/**
+ * @deprecated This provider is deprecated and will be removed in future releases.
```
Comment (bang9) [inline src/modules/Channel/context/ChannelProvider.tsx:166]: I think `next major update` is better than `future release`. (`future release` is too general to determine a specific time.)

**Comment by bang9** [comment]
Looks good to me! Let's wait for Irene's review.

**Review by bang9 — APPROVED** [review]

**Review by AhyoungRyu — APPROVED** [review]

**Pull request merged**